extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java

```diff
@@ -260,6 +220,79 @@ public void transform(TransformationContext ctx) {
         });
     }
 
+    @BuildStep
+    // Add reflective access to fallback methods, use bean index since fallback methods can be in bean classes
+    void processFallbackMethodsAndClases(BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<ReflectiveMethodBuildItem> reflectiveMethod) {
+        IndexView index = beanArchiveIndexBuildItem.getIndex();
+        Map<DotName, Set<String>> classesToScan = new HashMap<>();
+
+        for (AnnotationInstance annotation : index.getAnnotations(DotNames.FALLBACK)) {
+            AnnotationValue fallbackMethodValue = annotation.value("fallbackMethod");
+            if (fallbackMethodValue == null) {
+                continue;
+            }
+            String fallbackMethod = fallbackMethodValue.asString();
+
+            // @Fallback can only be present on methods, so this is just future-proofing
+            AnnotationTarget target = annotation.target();
+            final ClassInfo clazz = target.asMethod().declaringClass();
+            if (target.kind() != Kind.METHOD) {
+                continue;
+            }
+
+            // Scan both the hierarchy of the declaring class and its interfaces like in
+            // io.smallrye.faulttolerance.internal.SecurityActions.findDeclaredMethodNames
+            DotName name = clazz.name();
+            boolean isBean = beanArchiveIndexBuildItem.getGeneratedClassNames().contains(name);
```

Comment:
@Ladicek is this is the right way to do it?

Could we have non-generated beans as well or generated classes that are not actually beans? 

If so, how can I most effectively check if the DotName corresponds to a bean?

Reply:
I'm not sure why you'd need to do this. If you find a `@Fallback` annotation in the bean archive index, the declaring class is most likely a bean class (and if not, it is most likely a superclass of one). I wouldn't bother trying to figure out whether the class is a bean class or not.

Reply:
I am trying to keep the reflection registrations to the minimum and in theory we only need to register the superinterfaces of beans, so if something is not a bean (and also not a superclass of a bean) I am trying to skip registering its superinterfaces.

Is it safe to assume that having a non-bean class with a `@Fallback` annotated method is  **impossible** or at least **highly unlikely**?

I might be overthinking it, but I am not familiar with the code so please bear with me :)

Reply:
Is is technically possible to have a class with `@Fallback` that is not a bean class, but it's most likely a bug. I would classify that as _highly unlikely_, as you eloquently put it :-)

Reply:
Check removed PTAL